test/declaration_test.ts

```diff
@@ -255,6 +255,39 @@ class X {
   // @Ignore
   X.fakeConstructor$();
   external x();
+}`);
+    });
+    it('should emit extension methods to return Futures from methods that return Promises', () => {
+      expectTranslate(`declare interface MyMath {
+        randomInRange(start: number, end: number): Promise<number>;
+      }`).to.equal(`import "dart:async" show Completer;
+
+@anonymous
+@JS()
+abstract class MyMath {
+  external dynamic /*Promise<num>*/ randomInRange(num start, num end);
```

Comment:
why is Promise<num> commented out?
I would expect we should be emitting
@JS()
class Promise {
...
}